config/initializers/dfe-analytics.rb

```diff
@@ -0,0 +1,49 @@
+DfE::Analytics.configure do |config|
+  # Whether to log events instead of sending them to BigQuery.
+  #
+  config.log_only = false
+
+  # Whether to use ActiveJob or dispatch events immediately.
+  #
+  # config.async = true
+
+  # Which ActiveJob queue to put events on
+  #
+  config.queue = :big_query
+
+  # The name of the BigQuery table we’re writing to.
+  #
+  config.bigquery_table_name = 'events'
+
+  # The name of the BigQuery project we’re writing to.
+  #
+  config.bigquery_project_id = ENV['BIG_QUERY_PROJECT_ID']
+
+  # The name of the BigQuery dataset we're writing to.
+  #
+  config.bigquery_dataset = ENV['BIG_QUERY_DATASET']
+
+  # Service account JSON key for the BigQuery API. See
+  # https://cloud.google.com/bigquery/docs/authentication/service-account-file
+  #
+  config.bigquery_api_json_key = ENV['BIG_QUERY_API_JSON_KEY']
+
+  # Passed directly to the retries: option on the BigQuery client
+  #
+  # config.bigquery_retries = 3
+
+  # Passed directly to the timeout: option on the BigQuery client
+  #
+  # config.bigquery_timeout = 120
+
+  # A proc which returns true or false depending on whether you want to
+  # enable analytics. You might want to hook this up to a feature flag or
+  # environment variable.
+  #
+  config.enable_analytics = proc { FeatureFlag.active?(:send_request_data_to_bigquery) }
```

Comment:
Out of interest, why is this a proc? Wouldn't `FeatureFlag.active?(:send_request_data_to_bigquery)` also return true or false

Reply:
It is a proc because the block is saved to this enable_analytics config var so then to be called during request time and not when Rails is booting. So "save" the block here, call it later.
The method is defined here: https://github.com/DFE-Digital/dfe-analytics/blob/53efa7bfb759cdb1384133b11600eb041dfcbb03/lib/dfe/analytics.rb#L74

Reply:
ah, thats makes sense, the proc is stored in var to be called later by that method `self.enabled?`. Thanks!